internal/backend/storage/fs/store.go

```diff
@@ -60,6 +62,12 @@ func (s *Store) Set(ctx context.Context, name string, value []byte) error {
 	}
 	debug.Log("Writing %s to %s", name, filepath.Join(s.path, name))
 
+	oldvalue, err := os.ReadFile(filepath.Join(s.path, name))
```

Comment:
This is at the core of what this PR is about. I think this block deserves a small comment explaining why we're doing this.